experimental/Schemes/SpaceGerms.jl

```diff
@@ -48,52 +45,266 @@ A space germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X``
+      (ideal(parent(f),[f]) == modulus(OO(X))) || error("given f does not define given X")
+    end
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X,f)
+  end
+
+## the following is currently unused....
+## as no backend for groebner computations is currently available in this case
+  function HypersurfaceGerm(X::GermAtGeometricPoint, f::MPolyLocRingElem; check::Bool=true)
+    base_ring(modulus(OO(X))) == parent(f) || error("baserings do not match")
+    @check begin
+      (ideal(parent(f),[f]) == modulus(OO(X))) || error("given f does not define given X")
+    end
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X,f)
+  end
+end
+
+@doc raw"""
+    CompleteIntersectionGerm{BaseRingType, RingType, SpecType}
+A complete intersection germ ``(X,O_{(X,x)}``, i.e. a ringed space with underlying scheme ``X`` of type SpecType and local ring ``O_{(X,x)}`` of type `RingType` over some base ring ``k`` of type `BaseRingType`.
+"""
+@attributes mutable struct CompleteIntersectionGerm{BaseRingType<:Ring, RingType<:Ring, SpecType<:Spec} <: AbsSpaceGerm{BaseRingType, RingType}
+  X::SpecType
+  v::Vector{RingElem}
+
+  function CompleteIntersectionGerm(X::GermAtClosedPoint, v::Vector{T}; check::Bool=true) where T<:MPolyLocRingElem
+    R = base_ring(modulus(OO(X)))
+    all(x->parent(x) == R, v) || error("base_rings do not coincide")
+    @check begin
+## TODO: change dim(Spec(R)) to dim(R) as soon as the fixes for dim have
+## been moved to the algebra side!!!!
+      length(v) == dim(Spec(R)) - dim(X) || error("not a complete intersection")
+      modulus(OO(X)) == ideal(R,v) || error("given tuple does not generate modulus")
+    end
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X,v)
+  end
+
+## the following one is currently unused...
+## as no backend for groebner computations is currently available in this case
+  function CompleteIntersectionGerm(X::GermAtGeometricPoint, v::Vector{MPolyLocRingElem}; check::Bool=true)
+    R = base_ring(OO(X))
+    all(x->parent(x) == R, v) || error("base_rings do not coincide")
+    @check begin
+      length(v) == dim(Spec(R)) - dim(X) || error("not a complete intersection")
+      modulus(OO(X)) == ideal(R,v) || error("given tuple does not generate modulus")
+    end
+    return new{typeof(base_ring(X)), typeof(OO(X)), typeof(X)}(X,v)
+  end
+end
+
+##############################################################################
+## Some more shorthand notation
+##############################################################################
+AnySpaceGerm = Union{SpaceGerm, HypersurfaceGerm, CompleteIntersectionGerm}
+AnySpaceGermClosedPoint = Union{SpaceGerm{<:Ring,<:Ring,<:GermAtClosedPoint},
+                                HypersurfaceGerm{<:Ring,<:Ring,<:GermAtClosedPoint},
+                                CompleteIntersectionGerm{<:Ring,<:Ring,<:GermAtClosedPoint}}
+AnySpaceGermGeometricPoint = Union{SpaceGerm{<:Ring,<:Ring,<:GermAtGeometricPoint},
```

Comment:
```suggestion
const AnySpaceGerm = Union{SpaceGerm, HypersurfaceGerm, CompleteIntersectionGerm}
const AnySpaceGermClosedPoint = Union{SpaceGerm{<:Ring,<:Ring,<:GermAtClosedPoint},
                                HypersurfaceGerm{<:Ring,<:Ring,<:GermAtClosedPoint},
                                CompleteIntersectionGerm{<:Ring,<:Ring,<:GermAtClosedPoint}}
const AnySpaceGermGeometricPoint = Union{SpaceGerm{<:Ring,<:Ring,<:GermAtGeometricPoint},
```
If these are `const` then the compiler can optimize any references to these. If they are not.... well then not :-)